### PR TITLE
[CM H2] Third party licensing cleanup

### PR DESCRIPTION
**Pull request by a-kraschitzer**
## Basics

These points need to be fulfilled for every PR:

- [X] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [X] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [X] References to issues, e.g. `close #X`, are in the commit messages.
- [X] The buildservers are happy. If not, fix **in this order**:
  - [X] add a line in `doc/news/_preparation_next_release.md`
  - [X] reformat the code with `scripts/dev/reformat-all`
  - [X] make all unit tests pass
  - [X] fix all memleaks
- [X] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: scripts/check-third-party-license-file-paths [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+for i in `grep -oP "(?<=Files: |\s{7})([^\*].*/.*)" ../doc/THIRD-PARTY-LICENSES`
+do
+	echo -n "../$i "
+	eval ls -d "../$i" 2> /dev/null | wc -l
```
Comment (a-kraschitzer) [inline scripts/check-third-party-license-file-paths:6]: @markus2330 you mentioned that this script produces false positives?
How exactly? I was not able to reproduce any false positives.

File: doc/THIRD-PARTY-LICENSES [hunk 2]
```diff
@@ -54,44 +48,44 @@ License: public-domain
...
+Files: src/tools/website/resources/assets/skin/base/fonts/font-awesome/*
 Copyright: 2016 Dave Gandy <dave@fontawesome.io>
 License: MIT
 
 Files: src/tools/qt-gui/src/markdownconverter/*
 Copyright: 2013 Christian Loose
 License: BSD-3-clause-markdownconverter
 
-Files: scripts/realpath
+Files: scripts/dev/realpath
```
Comment (a-kraschitzer) [inline doc/THIRD-PARTY-LICENSES:79]: I added further changes that were not explicitly asked for where I found invalid paths

File: scripts/check-third-party-license-file-paths (hunk 1 above)
Comment (markus2330) [reply scripts/check-third-party-license-file-paths:5]: I think it was related to wildcards.

Comment (a-kraschitzer) [reply scripts/check-third-party-license-file-paths:5]: Alright!

The supported formatting is still quite limited, but I guess we can keep the non-path results to a minimum.

**Comment by a-kraschitzer** [comment]
@markus2330 I don't know why the bashisms test is failing. I tried 
```
$ checkbashisms check-third-party-license-file-paths
could not find any possible bashisms in bash script check-third-party-license-file-paths
```
Please advise.

**Comment by markus2330** [comment]
I agree this is unrelated to your script. This looks to me like the "you may get strange results" suddenly bit us. Maybe it is already resolved, @mpranj has done some fixes for exclusion of scripts. We probably need to exclude the files which are neither scripts nor have a shebang, I opened #3834.

File: scripts/check-third-party-license-file-paths [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (mpranj) [inline scripts/check-third-party-license-file-paths:1]: Sorry, but in this case it works as expected.

From `checkbashisms` help:
```
This script performs basic checks for the presence of bashisms
in /bin/sh scripts and the lack of bashisms in /bin/bash ones.
```

The new script uses **no** bashisms, so there is **no** need to depend on bash. It should work with any POSIX shell, so we can use `/bin/sh` here.

```suggestion
#!/bin/sh
```

Also it looks like this script itself generates some false positive files:
```
../documentation 0
../and/or 0
../other 0
../materials 0
../provided 0
../with 0
../the 0
../distribution. 0
```

Comment (a-kraschitzer) [reply scripts/check-third-party-license-file-paths]: Thank you, pushed the changes

**Comment by markus2330** [comment]
Thank you so much for these fixes!

**Pull request merged**